### ISSUE TITLE
hcp_vault_cluster - updates for plus tier

### ISSUE THREAD

**Issue by assareh** [hashicorp/terraform-provider-hcp#256]
### Description

Now that [Plus tier](https://www.hashicorp.com/blog/multi-region-replication-now-available-with-hcp-vault) clusters are available, there are some updates needed.

Based on my brief test the `hcp_vault_cluster` resource already supports creation of plus tier clusters, however the docs should be updated to reflect this. The `hcp_vault_cluster` (data source) should also be updated. 

Support for provisioning a replica cluster should be added.

### New or Affected Resource(s)

* hcp_vault_cluster (resource)
* hcp_vault_cluster (data source)

### Potential Terraform Configuration

```hcl
resource "hcp_vault_cluster" "example" {
  cluster_id = "vault-cluster"
  hvn_id     = hcp_hvn.example.hvn_id
  tier       = "plus_small"
}

resource "hcp_vault_cluster" "example-replica" {
  cluster_id         = "vault-cluster-replica"
  primary_cluster_id = hcp_vault_cluster.example.id
  hvn_id             = hcp_hvn.replica.hvn_id
  tier               = "plus_small"
}
```

### References

* #0000

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by bcmdarroch** [comment]
Hi @assareh, thanks for submitting the issue! We'll work with our HCP Vault team to get this in.

**Comment by bcmdarroch** [comment]
This functionality has been made available in v0.24.0! 🎉
https://github.com/hashicorp/terraform-provider-hcp/releases/tag/v0.24.0
